ast/src/parsed/pattern.rs

```diff
@@ -0,0 +1,432 @@
+use powdr_number::BigInt;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use std::iter::{empty, once};
+
+use super::{asm::SymbolPath, visitor::Children};
+
+/// When a pattern is specialised, in some cases the algorithm will return
```

Comment:
Maybe it makes sense to move this a bit down so that Pattern is the first element in this file?